adapters/pixfuture/pixfuture.go

```diff
@@ -0,0 +1,139 @@
+	}
+
+	var errs []error
+	var adapterRequests []*adapters.RequestData
+
+	for i := range request.Imp {
+		imp := &request.Imp[i]
+
+		// Log raw imp.Ext for debugging
+		var bidderExt adapters.ExtImpBidder
+		if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: err.Error(),
+			}}
+		}
+
+		var ext openrtb_ext.ImpExtPixfuture
+		if err := jsonutil.Unmarshal(bidderExt.Bidder, &ext); err != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: err.Error(),
+			}}
+		}
+
+		reqJSON, err := jsonutil.Marshal(request)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		headers := http.Header{}
+		headers.Set("Content-Type", "application/json")
+		headers.Set("Accept", "application/json")
+
+		adapterRequests = append(adapterRequests, &adapters.RequestData{
+			Method:  http.MethodPost,
+			Uri:     a.endpoint,
+			Body:    reqJSON,
+			Headers: headers,
+			ImpIDs:  []string{imp.ID},
+		})
+	}
+
+	if len(adapterRequests) == 0 && len(errs) > 0 {
+		return nil, errs
+	}
+	return adapterRequests, errs
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{Message: "Unexpected status code: " + strconv.Itoa(response.StatusCode)}}
+	}
+
+	var bidResp openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{Message: "Invalid response format: " + err.Error()}}
+	}
+
+	bidResponse := adapters.NewBidderResponse()
+	bidResponse.Currency = bidResp.Cur
+
+	var errs []error
+	for i := range bidResp.SeatBid {
+		seatBid := &bidResp.SeatBid[i]
+		for j := range seatBid.Bid {
+			bid := &seatBid.Bid[j]
+			bidType, err := getMediaTypeForBid(*bid)
+			if err != nil {
+				errs = append(errs, &errortypes.BadServerResponse{Message: "Failed to parse impression \"" + bid.ImpID + "\" mediatype"})
+				continue
+			}
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     bid,
+				BidType: bidType,
+			})
+		}
+	}
+
+	if len(bidResponse.Bids) == 0 {
+		if len(errs) > 0 {
+			return nil, errs
+		}
+		return nil, nil
+	}
+	return bidResponse, errs
+}
+
+func getMediaTypeForBid(bid openrtb2.Bid) (openrtb_ext.BidType, error) {
+	var ext struct {
+		Prebid struct {
+			Type string `json:"type"`
+		} `json:"prebid"`
+	}
+	if err := jsonutil.Unmarshal(bid.Ext, &ext); err != nil {
+		return "", err
+	}
+
+	switch ext.Prebid.Type {
+	case "banner":
+		return openrtb_ext.BidTypeBanner, nil
+	case "video":
+		return openrtb_ext.BidTypeVideo, nil
+	case "native":
+		return openrtb_ext.BidTypeNative, nil
```

Comment:
Given that `static/bidder-info/pixfuture.yaml` doesn't list `native` as a supported type, no `request.Imp` elements with a non-nil `request.Imp[i].Native` will get to `MakeRequests()` and no native-media requests will be sent to your bid server. Do you want to list `native` as a supported media type in `static/bidder-info/pixfuture.yaml`?
```
 5   capabilities:
 6     site:
 7       mediaTypes:
 8         - banner
 9         - video
   +       - native
10     app:
11       mediaTypes:
12         - banner
13         - video
   +       - native
14   userSync:
static/bidder-info/pixfuture.yaml
```

Reply:
It looks like you're dealing with some discrepancies in your Git repository. 
From your message, it seems like you've made a change to pixfuture.yaml, 
but there’s confusion about coverage.out, and you've frequently rolled back go.mod and go.sum using patch checkout.

Reply:
> From your message, it seems like you've made a change to pixfuture.yaml,

I didn't. The block of code in my comment above was just meant to illustrate that if you decide to modify file `static/bidder-info/pixfuture.yaml` by adding those `native` entries, Prebid Server Core will not filter out requests that come with native format. I see that in your last update you addressed this comment. Thank you.